storage/store.go

```diff
@@ -1822,3 +1822,21 @@ func (s *Store) PublishStatus() error {
 func (s *Store) SetRangeRetryOptions(ro retry.Options) {
 	s.ctx.RangeRetryOptions = ro
 }
+
+// hasReplicaExist used to judgement if "rngDesc" has overlap with ranges
+// in "replicasByKey" btree, it is not thread saft.
+func (s *Store) hasOverlapReplica(rngDesc *roachpb.RangeDescriptor) bool {
```

Comment:
>
// hasOverlappingReplicaLocked returns true if a Replica overlapping the given descriptor is present on the Store.